src/groups/general_unitary_groups.jl

```diff
@@ -0,0 +1,262 @@
+@doc raw"""
+    GeneralUnitaryMultiplicationGroup{n,𝔽,M} = GroupManifold{𝔽,M,MultiplicationOperation}
+
+A generic type for Lie groups based on a unitary property and matrix multiplcation,
+see e.g. [`Orthogonal`](@ref), [`SpecialOrthogonal`](@ref), [`Unitary`](@ref), and [`SpecialUnitary`](@ref)
+"""
+struct GeneralUnitaryMultiplicationGroup{n,𝔽,S} <: AbstractDecoratorManifold{𝔽}
+    manifold::GeneralUnitaryMatrices{n,𝔽,S}
+end
+
+@inline function active_traits(f, ::GeneralUnitaryMultiplicationGroup, args...)
+    if is_metric_function(f)
+        #pass to Rotations by default - but keep Group Decorator for the retraction
+        return merge_traits(
+            IsGroupManifold(MultiplicationOperation()),
+            IsExplicitDecorator(),
+        )
+    else
+        return merge_traits(
+            IsGroupManifold(MultiplicationOperation()),
+            HasBiinvariantMetric(),
+            IsDefaultMetric(EuclideanMetric()),
+            IsExplicitDecorator(), #pass to the inner M by default/last fallback
+        )
+    end
+end
+
+function allocate_result(
+    ::GeneralUnitaryMultiplicationGroup,
+    ::typeof(exp),
+    ::Identity{MultiplicationOperation},
+    X,
+)
+    return allocate(X)
+end
+function allocate_result(
+    ::GeneralUnitaryMultiplicationGroup,
+    ::typeof(log),
+    ::Identity{MultiplicationOperation},
+    q,
+)
+    return allocate(q)
+end
+
+decorated_manifold(G::GeneralUnitaryMultiplicationGroup) = G.manifold
+
+@doc raw"""
+     exp_lie(G::Orthogonal{2}, X)
+     exp_lie(G::SpecialOrthogonal{2}, X)
+
+Compute the Lie group exponential map on the [`Orthogonal(2)`] group.
```

Comment:
```suggestion
Compute the Lie group exponential map on the [`Orthogonal(2)`](@ref) or [`SpecialOrthogonal(2)`](@ref) group.
```
I'm not sure if `@ref` will work with that `(2)`. If not, I'd put `(2)` after the link, like in `exp_lie(::GeneralUnitaryMultiplicationGroup{4,ℝ}, X)` docstring below.

Reply:
I think one has to use the parameters then, so probably `[`Orthgonal(2)`](@ref Orthogonal{2})`? But we could try this one as well first.

Reply:
```suggestion
Compute the Lie group exponential map on the [`Orthogonal`](@ref)`(2)` or [`SpecialOrthogonal`](@ref)`(2)` group.
```

what about this variant?

Reply:
Looks fine :+1: .